src/JSInterop/Microsoft.JSInterop/src/Properties/AssemblyInfo.cs

```diff
@@ -0,0 +1,3 @@
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("Microsoft.JSInterop.WebAssembly, PublicKey=0024000004800000940000000602000000240000525341310004000001000100f33a29044fa9d740c9b3213a93e57c84b472c84e0b8a0e1ae48e67a9f8f6de9d5f7f3d52ac23e48ac51801f1dc950abe901da34d2a9e3baadb141a17c77ef3c565dd5ee5054b91cf63bb3c6ab83f72ab3aafe93d0fc3c2348b764fafb0b1c0733de51459aeab46580384bf9d74c4e28164b7cde247f891ba07891c9d872ad2bb")]
```

Comment:
The IVT here is my one main concern with this PR. We try to avoid them where possible because it suggests the design isn't really as extensible as we planned, and that the assembly factoring is a bit of a lie. Third parties don't have the benefit of being able to do this like we can.

Maybe it's the only possible solution, in which case OK. I was going to try to come up with a suggestion about how the IVT could be avoided but then noticed [this](https://github.com/dotnet/aspnetcore/pull/25548/files#r482979716) which might change the picture a bit anyway. Once that is resolved perhaps it would be a better time to try figuring out whether/how the IVT could be avoided. Do you have any ideas?

Reply:
I'll spend some time investigating this - the main reason for the change was so that `WebAssemblyJSObjectReference` could extend `JSInProcessObjectReference`, which is `internal` to `Microsoft.JSInterop`. Maybe we could make `IJSUnmarshalledObjectReference` not extend the `IJSInProcessObjectReference` interface, and likewise not have `WebAssemblyJSObjectReference` extend `JSInProcessObjectReference`. You would only get the unmarshalled interop methods when using an `IJSUnmarshalledObjectReference`, but maybe that's ok if developers are likely to use `IJSUnmarshalledObjectReference` instances only for unmarshalled interop calls.